reV/SAM/generation.py

```diff
@@ -624,6 +625,256 @@ def default(self):
+    def __init__(self, resource=None, meta=None, parameters=None,
+                 output_request=None):
+        """Initialize a SAM solar water heating object.
+
+        """
+        self.timezone = None
+        super().__init__(resource=resource, meta=meta, parameters=parameters,
+                         output_request=output_request)
+
+    def set_nsrdb(self, resource):
+        """
+        Set NSRDB resource file. Overloads Solar.set_nsrdb(). Solar thermal
+        PySAM models require a data file, not raw data.
+
+        Parameters
+        ----------
+        resource : pd.DataFrame
+            2D table with resource data. Available columns must have var_list.
+        """
+        self.time_interval = self.get_time_interval(resource.index.values)
+        # Create solar resource file
+        self._pysam_w_fname = self._create_pysam_wfile(self._meta, resource)
+        # pylint: disable=E1101
+        self[self._pysam_weather_tag] = self._pysam_w_fname
+
+    def _create_pysam_wfile(self, meta, resource, drop_leap=True):
+        """
+        Create PySAM weather input file.
+
+        Parameters
+        ----------
+        meta : pd.DataFrame
+            1D table with resource meta data.
+        resource : pd.DataFrame
+            2D table with resource data. Available columns must have var_list.
+        drop_leap : bool
+            Drops February 29th from the resource data. If False, December
+            31st is dropped from leap years.
+        """
+        fname = f'{self._site}_weather.csv'
+        logger.debug(f'Creating PySAM weather data file: {fname}')
+
+        # Process metadata
+        m = pd.DataFrame(meta).T
+        self.timezone = m.timezone
+        m['Source'] = 'NSRDB'
+        m['Location ID'] = meta.name
+        m['City'] = '-'
+        m['State'] = m.state.apply(lambda x: '-' if x == 'None' else x)
+        m['Country'] = m.country.apply(lambda x: '-' if x == 'None' else x)
+        m['Latitude'] = m.latitude
+        m['Longitude'] = m.longitude
+        m['Time Zone'] = m.timezone
+        m['Elevation'] = m.elevation
+        m['Local Time Zone'] = m.timezone
+        m['Local Time Zone'] = m.timezone
+        m['Dew Point Units'] = 'c'
+        m['DHI Units'] = 'w/m2'
+        m['DNI Units'] = 'w/m2'
+        # m['GHI Units'] = 'w/m2'
+        m['Temperature Units'] = 'c'
+        m['Pressure Units'] = 'mbar'
+        # m['Wind Direction Units'] = 'Degrees'
+        m['Wind Speed'] = 'm/s'
+        # m['Surface Albedo Units'] = 'N/A'
+        # m['Version'] = 'v3.0.1'
+        m = m.drop(['elevation', 'timezone', 'country', 'state', 'county',
+                    'urban', 'population', 'landcover', 'latitude',
+                    'longitude'], axis=1)
+        m.to_csv(fname, index=False, mode='w')
+
+        # Process data
+        df = resource.copy()
+        df['dt'] = df.index
+
+        # Drop days for leap year before adjusting timezone
+        if df.dt.dt.is_leap_year.all():
```

Comment:
This is pretty messy. Use the SAM.drop_leap() static method instead. Example from reV_run():

```
        # Use resource object iterator
        for res_df, meta in resources:

            # drop the leap day
            if drop_leap:
                res_df = cls.drop_leap(res_df)